apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c

```diff
@@ -74,241 +63,754 @@ static void clean_all_data(int fd, char *buffer)
 * @brief            open and check pcm is ready or not
 * @scenario         Open pcm and check its state
 * @apicovered       pcm_is_ready
 * @precondition     pcm_open should be called
 * @postcondition    NA
 */
-static void itc_audio_pcm_is_ready_tc_p(void)
+static void itc_audio_pcm_is_ready_p(void)
+{
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+	TC_ASSERT_EQ_CLEANUP("pcm_is_ready", pcm_is_ready(g_pcm), 1, pcm_close(g_pcm));
+
+	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_is_ready_n_after_close
+* @brief            open and check pcm is ready or not after pcm close
+* @scenario         Open and close pcm and check its state
+* @apicovered       pcm_is_ready
+* @precondition     pcm_open and pcm_close should be called
+* @postcondition    NA
+*/
+static void itc_audio_pcm_is_ready_n_after_close(void)
 {
+	int ret = 0;
+	g_pcm = NULL;
 	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
-	TC_ASSERT_GT_CLEANUP("pcm_is_ready", pcm_is_ready(g_pcm), 0, pcm_close(g_pcm));
+
 	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+
+	ret = pcm_is_ready(g_pcm);
+	TC_ASSERT_EQ("pcm_is_ready", ret, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_is_ready_n
+* @brief            open and check pcm is ready or not
+* @scenario         Open pcm and check its state with NULL pcm
+* @apicovered       pcm_is_ready
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_audio_pcm_is_ready_n(void)
+{
+	TC_ASSERT_EQ("pcm_is_ready", pcm_is_ready(NULL), 0);
 	TC_SUCCESS_RESULT();
 }
 
 /**
-* @testcase         audio_pcm_prepare_p
+* @testcase         itc_audio_pcm_prepare_p
 * @brief            open and prepare pcm
 * @scenario         Open pcm and prepare
 * @apicovered       pcm_prepare
 * @precondition     pcm_open should be called
 * @postcondition    NA
 */
-static void itc_audio_pcm_prepare_tc_p(void)
+static void itc_audio_pcm_prepare_p(void)
 {
+	g_pcm = NULL;
 	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
 	TC_ASSERT_EQ_CLEANUP("pcm_prepare", pcm_prepare(g_pcm), 0, pcm_close(g_pcm));
+
+	TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_audio_pcm_prepare_n_after_close
+* @brief            open and prepare pcm
+* @scenario         Open and close pcm and prepare
+* @apicovered       pcm_prepare
+* @precondition     pcm_open and pcm_close should be called
+* @postcondition    NA
+*/
+static void itc_audio_pcm_prepare_n_after_close(void)
+{
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
```

Comment:
cleanup needed here.

Reply:
There is no need to close g_pcm, as it is not yet open. This Assert checks whether pcm_open API is actually open or not, if it is not opened, then it cannot be closed